packages/core/core/src/EntryResolver.js

```diff
@@ -0,0 +1,104 @@
+// @flow
+import type {FileSystem} from '@parcel/fs';
+import type {FilePath, File} from '@parcel/types';
+import type {ParcelOptions} from './types';
+import path from 'path';
+import {isGlob, glob} from '@parcel/utils';
+
+type EntryResult = {|
+  entries: Array<FilePath>,
+  files: Array<File>
+|};
+
+export class EntryResolver {
+  fs: FileSystem;
+
+  constructor(options: ParcelOptions) {
+    this.fs = options.inputFS;
+  }
+
+  async resolveEntry(entry: FilePath): Promise<EntryResult> {
+    if (isGlob(entry)) {
+      let files = await glob(entry, this.fs, {
+        absolute: true,
+        onlyFiles: false
+      });
+      let results = await Promise.all(files.map(f => this.resolveEntry(f)));
+      return results.reduce(
+        (p, res) => ({
+          entries: p.entries.concat(res.entries),
+          files: p.files.concat(res.files)
+        }),
+        {entries: [], files: []}
+      );
+    }
+
+    let stat;
+    try {
+      stat = await this.fs.stat(entry);
+    } catch (err) {
+      throw new Error(`Entry ${entry} does not exist`);
+    }
+
+    if (stat.isDirectory()) {
+      let pkg = await this.readPackage(entry);
+      if (pkg && typeof pkg.source === 'string') {
```

Comment:
Internally, we are currently declaring a source in the `pkg` as well but under a custom property. Do we expect to make this property standard, or should we allow configuration on the name of this key?

Reply:
`source` is already supported by Parcel 1 and is in the RFC for v2 as a "standardish" field: https://github.com/parcel-bundler/parcel/blob/master/PARCEL_2_RFC.md#sources